lib/jira/resource/user.rb

```diff
@@ -13,8 +13,8 @@ def self.singular_path(client, key, prefix = '/')
 
       # Cannot retrieve more than 1,000 users through the api, please see: https://jira.atlassian.com/browse/JRASERVER-65089
       def self.all(client)
-        search_token = client.cloud_instance? ? "_" : "@"
-        response  = client.get("/rest/api/2/user/search?username=#{search_token}&maxResults=#{MAX_RESULTS}")
+        search_param = client.cloud_instance? ? "query=" : "username=@"
```

Comment:
When we tried this query we would get inconsistent results, which makes me a bit worried about this solution. Although, we could have gotten inconsistent results due to JIRA API migrations. Have we tested this call with multiple JIRA accounts?

Reply:
Testing the above change this morning, I got:
- 245 for Luminos (affected by GDPR change)
- 44 for Denali (affected by GDPR change)
- 100 for Appsbroker (affected by GDPR change)
- 605 for Phenom People (not affected by GDPR), compared to 605 via `User.all` which is still working for them